### ISSUE TITLE
The path in the error message on Windows contains multiple separators

### ISSUE THREAD

**Issue by ahaoboy** [rust-lang/rust-clippy#13078]
### Summary

![image](https://github.com/rust-lang/rust-clippy/assets/19884146/3c7654a9-bf6f-4f48-b8bb-dcd7d8ae85fe)

```
warning: unused variable: `a`
 --> mpv-easy-ext\./src/main.rs:2:9
  |
2 |     let a = 1;
  |         ^ help: if this is intentional, prefix it with an underscore: `_a`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: `mpv-easy-ext` (bin "mpv-easy-ext") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.19s
```

### Reproducer

I can’t reproduce this bug in a new cargo project.

```rust
git clone https://github.com/mpv-easy/mpv-easy.git
```

./mpv-easy/mpv-easy-ext/src/main.rs
```rust
fn main() {
    let a = 1;
}
```
```
cd  ./mpv-easy/mpv-easy-ext
 cargo clippy
```

I expected to see this happen:

```
warning: unused variable: `a`
 --> mpv-easy-ext/src/main.rs:2:9
```

Instead, this happened:
```
warning: unused variable: `a`
 --> mpv-easy-ext\./src/main.rs:2:9
```

### Version

```text
rustc 1.79.0 (129f3b996 2024-06-10)
binary: rustc
commit-hash: 129f3b9964af4d4a709d1383930ade12dfe7c081
commit-date: 2024-06-10
host: x86_64-pc-windows-gnu
release: 1.79.0
LLVM version: 18.1.7
```

**Comment by Alexendoo** [comment]
It's because of the specified `path` in

```rust
[[bin]]
name = "mpv-easy-ext"
path = "./src/main.rs"
```

It would be a cargo issue that happens with `cargo check`/`cargo build` too so worth opening an issue there, I couldn't find an existing one from a quick search

**Comment by Alexendoo** [comment]
Closing in favour of the Cargo issue
